app/models/result.rb

```diff
@@ -126,8 +126,9 @@ def unrelease_partial_results
   end
 
   def check_for_nil_marks
-    num_criteria = submission.assignment.rubric_criteria.count +
-                   submission.assignment.flexible_criteria.count
+    peer_review = submission.assignment.is_peer_review?
+    num_criteria = filter_criteria(submission.assignment.rubric_criteria, peer_review).count +
```

Comment:
Use the `get_criteria` method in the Assignment model rather than this "filter_criteria".